packages/eslint-plugin-pf-codemods/lib/rules/v5/datalist-remove-selectableRow.js

```diff
@@ -1,25 +1,173 @@
-const { getFromPackage} = require('../../helpers');
+const { getFromPackage, findVariableDeclaration } = require("../../helpers");
 
 //https://github.com/patternfly/patternfly-react/pull/8827
 module.exports = {
-  meta: {},
+  meta: { fixable: "code" },
   create: function (context) {
-    const dataListImport =  getFromPackage(context, '@patternfly/react-core')
-      .imports.filter(specifier => specifier.imported.name == 'DataList');
+    const dataListImport = getFromPackage(
+      context,
+      "@patternfly/react-core"
+    ).imports.find((specifier) => specifier.imported.name == "DataList");
 
-    return dataListImport.length === 0 ? {} : {
-      JSXOpeningElement(node) {
-        if (dataListImport.map(imp => imp.local.name).includes(node.name.name)) {
-          const selectableRow = node.attributes.find(a => a.name?.name === 'selectableRow');
-          if (selectableRow) {
-            context.report({
-              node,
-              message: `DataList's selectableRow property has been replaced with onSelectableRowChange. The order of the params in the callback has also been updated so that the event param is first.`,
-            });
-          }
+    if (!dataListImport) {
+      return {};
+    }
+
+    let rangeStartsToReplaceConst = [];
```

Comment:
Are there instances where this will ever have a length larger than 1, or have duplicate items where the `new Set` on line 161 is needed?

Reply:
Yes, this can have length larger than one. It is filled every time when we are changing the value of `.onChange` prop of the `selectableRowObject` (so in a `MemberExpression`). And because we are changing the original variable `const` object to a function, we have to change it to `let`, as then we assing a new value not to the `.onChange` prop, but to the function variable directly.

Example:
```
const onChange = (id) => {};
function onChangeFunc(id, event) {}

const selectableRowObject = { onChange: function (id, event) {} };
selectableRowObject.onChange = onChangeFunc;
selectableRowObject["onChange"] = onChange;
selectableRowObject["onChange"] = onChangeFunc;

const selectableRowObject2 = { onChange: function (id, event) {} };
selectableRowObject2.onChange = onChangeFunc;

<>
  <DataList selectableRow={selectableRowObject} />
  <DataList selectableRow={selectableRowObject2} />
</>;
```
console.log():
```
rangeStartsToReplaceConst
[ 150, 150, 150, 355 ]
```
without deduplication with the `new Set` it throws an error, as we are trying to replace it more times.

There will probably not be many (maybe none) cases, where someone uses the selectableRowObject like that and reassigns the value, but just to be sure